src/BizHawk.Client.EmuHawk/MainForm.Events.cs

```diff
@@ -538,6 +539,15 @@ private void SynclessRecordingMenuItem_Click(object sender, EventArgs e)
 		private void CaptureOSDMenuItem_Click(object sender, EventArgs e)
 		{
 			Config.AviCaptureOsd ^= true;
+			if (Config.AviCaptureOsd)
+				Config.AviCaptureLua = true;
+		}
+
+		private void CaptureLuaMenuItem_Click(object sender, EventArgs e)
+		{
+			Config.AviCaptureLua = !CaptureLuaMenuItem.Checked;
+			if (!Config.AviCaptureLua)
+				Config.AviCaptureOsd = false;

```

Comment:
Is this trying to re-implement the behaviour of a checkbox with a child checkbox? I can't understand what the changes in this file do.

Reply:
"checkbox with a child checkbox" .. I guess? It's to avoid having Capture OSD still turned on when Capture Lua is turned off (because I haven't implemented a way to capture OSD elements without also capturing Lua).

Reply:
To allow OSD + Lua, ~OSD + Lua, and ~OSD + ~Lua, but not OSD + ~Lua, try:
```csharp
private void AVSubMenu_DropDownOpened(object sender, EventArgs e)
{
	// ...
	CaptureOSDMenuItem.Checked = Config.AviCaptureOsd;
	CaptureLuaMenuItem.Checked = Config.AviCaptureLua;
	// ...
}

private void CaptureOSDMenuItem_Click(object sender, EventArgs e)
{
	Config.AviCaptureOsd = ((CheckBox) sender).Checked;
}

private void CaptureLuaMenuItem_Click(object sender, EventArgs e)
{
	var c = ((CheckBox) sender).Checked;
	Config.AviCaptureLua = c;
	CaptureOSDMenuItem.Enabled = c;
	if (!c) CaptureOSDMenuItem.Checked = false; // on negative edge only; setter triggers event handler which sets config
}
```

Reply:
With this code, if config is currently ~OSD + ~Lua and the user checks CaptureOSD, then Config will have OSD + ~Lua.

Also, imo disabling the OSD menu item when ~Lua is less user-friendly than current implementation (which is to automatically turn on Lua when the user turns on OSD).
(We would also need to add code to conditionally en/disable CaptureOSDMenuItem inside AVSubMenu_DropDownOpened or on init if we go that route, for consistency.)

Perhaps some comments added to current implementation would help?
`			Config.AviCaptureLua = !CaptureLuaMenuItem.Checked;
			if (!Config.AviCaptureLua) // Logic to capture OSD w/o Lua does not currently exist, so disallow that.
				Config.AviCaptureOsd = false;
`

Reply:
If you used the CheckBox' state instead of writing to and then immediately reading from config, then added comments, that would also work.